docs/topics/request-response.rst

```diff
@@ -312,6 +312,143 @@ errors if needed::
                 request = failure.request
                 self.logger.error('TimeoutError on %s', request.url)
 
+
+.. _request-fingerprints:
+
+Request fingerprints
+--------------------
+
+There are some aspects of scraping, such as filtering out duplicate requests
+(see :setting:`DUPEFILTER_CLASS`) or caching responses (see
+:class:`~scrapy.downloadermiddlewares.httpauth.HttpAuthMiddleware`), where you
+need the ability to generate a short, unique identifier from a
+:class:`~scrapy.http.Request` object: a request fingerprint.
+
+You often do not need to worry about request fingerprints, the default request
+fingerprinter works for most projects.
+
+However, there is no universal way to generate a unique identifier from a
+request, because different situations require comparing requests differently.
+For example, sometimes you may need to compare URLs case-insensitively, include
+URL fragments, exclude certain URL query parameters, include some or all
+headers, etc.
+
+To change how request fingerprints are built for your requests, use the
+:setting:`REQUEST_FINGERPRINTER_CLASS` setting.
+
+.. setting:: REQUEST_FINGERPRINTER_CLASS
+
+REQUEST_FINGERPRINTER_CLASS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Default: ``'scrapy.utils.request.RequestFingerprinter'``
+
+The import path of a :ref:`request fingerprinter
+<custom-request-fingerprinter>`.
+
+The default request fingerprinter takes into account a canonical version
+(:func:`w3lib.url.canonicalize_url`) of
+:attr:`request.url <scrapy.http.Request.url>` and the values of
+:attr:`request.method <scrapy.http.Request.method>` and
+:attr:`request.body <scrapy.http.Request.body>`. It then generates an
```

Comment:
```suggestion
:attr:`request.body <scrapy.http.Request.body>`. It then generates a
```
Nitpick

Reply:
Actually, I think “an” is correct here.

The use of “a” or “an” [is based on phonetics](https://writingexplained.org/a-vs-an-difference) rather than letters. Assuming that “SHA1” is pronounced by reading each letter and number separately, and because “s” as a standalone letter is pronounced [“ess”](https://en.wikipedia.org/wiki/S#cite_ref-2), I believe it should be preceded by “an”.

Reply:
Interesting :thinking: It still looks a bit weird, but I'm no native English speaker and I have no grounds to say otherwise. Thanks for the links!